File: Assignment 1/CountingSort/CountingSort.py
import sys
from time import time

def countSort(arr): 

	maximo = max(arr) + 1
	output = [0 for i in range(len(arr))] 
	count = [0 for i in range(maximo)] 


	for i in arr: 
		count[i] += 1

	for i in range(1,maximo): 
		count[i] += count[i-1] 

	for i in range(len(arr)-1, -1, -1): 
		output[count[arr[i]]-1] = arr[i] 
		count[arr[i]] -= 1

	for i in range(len(arr)): 
		arr[i] = output[i] 
	return arr 

#input de datos
archivo = sys.stdin.readlines()
n = int(archivo[0])
arr = []

dato = archivo[1].split()
for i in range (n):
    arr.append(int(dato[i]))

#medir tiempos
start = time()
countSort(arr)
finish = time()

print(finish-start,end = '')

File: Assignment 1/CountingSort/CountingSort.java
import java.util.Arrays;
import java.util.Scanner;

public class Counting {

      public static int max(int arr[]){
            int maximo = arr[0];

            for(int i = 1; i < arr.length; ++i)
                  if(maximo < arr[i])
                        maximo = arr[i];

            return maximo;
      }

      public static void sort(int arr[]) {
            int n = arr.length;
            int maximo = max(arr) + 1;

            int output[] = new int[n];

            int count[] = new int[maximo];

            for (int i = 0; i < maximo; ++i)
                  count[i] = 0;

            for (int i = 0; i < n; ++i)
                  ++count[arr[i]];

            for (int i = 1; i < maximo; ++i)
                  count[i] += count[i - 1];

            for (int i = n - 1; i >= 0; i--) {
                  output[count[arr[i]] - 1] = arr[i];
                  --count[arr[i]];
            }

            for (int i = 0; i < n; ++i)
                  arr[i] = output[i];
      }

      public static void main(String args[]) {

            Scanner scan = new Scanner(System.in);
            int input,size;
            size = scan.nextInt();

            int [] array = new int[size];
            for (int i = 0; i < size; i++){
                  input = scan.nextInt();
                  array[i] = input;
            }
            
            float tinicio, tfinal, tiempo;

            tinicio = System.nanoTime();
            sort(array);
            tfinal = System.nanoTime();
            tiempo = tfinal -tinicio;
            System.out.print(tiempo/1000000000);
      }
}
